chatkit-core/src/main/kotlin/com/pusher/chatkit/ChatkitDependencies.kt

```diff
@@ -1,13 +1,34 @@
 package com.pusher.chatkit
 
-import com.pusher.chatkit.pushnotifications.PushNotificationsFactory
+import com.pusher.SdkInfo
+import com.pusher.platform.MediaTypeResolver
 import com.pusher.platform.PlatformDependencies
+import com.pusher.platform.logger.Logger
 import com.pusher.platform.tokenProvider.TokenProvider
-import okhttp3.OkHttpClient
+import java.io.File
 
-interface ChatkitDependencies : PlatformDependencies {
-    val tokenProvider: TokenProvider
-    val okHttpClient: OkHttpClient?
-    val appHooks: AppHookEmitter
-    val pushNotifications: PushNotificationsFactory?
+open class ChatkitDependencies(
+    val tokenProvider: TokenProvider,
+    override val logger: Logger = EmptyLogger(),
+    override val mediaTypeResolver: MediaTypeResolver = NullMediaTypeResolver(),
+    override val sdkInfo: SdkInfo = SdkInfo(
+        product = "Chatkit",
+        sdkVersion = SDK_VERSION,
+        platform = "JVM",
+        language = "Kotlin"
+    )
+) : PlatformDependencies
+
+open class EmptyLogger : Logger {
+    override fun debug(message: String, error: Error?) { /* nop */ }
+    override fun error(message: String, error: Error?) { /* nop */ }
+    override fun info(message: String, error: Error?) { /* nop */ }
+    override fun verbose(message: String, error: Error?) { /* nop */ }
+    override fun warn(message: String, error: Error?) { /* nop */ }
+}
+
+class NullMediaTypeResolver : MediaTypeResolver {
+    override fun fileMediaType(file: File): String? = null
 }
+
+const val SDK_VERSION = "2.0.0-alpha1" // should match VERSION_NAME in parent's gradle.properties
```

Comment:
can we not use that gradle VERSION_NAME here? If not, we'll need to ensure our release process documents that this file will need to be updated.

Reply:
I provided more insight in [the commit log](https://github.com/pusher/chatkit-android/pull/195/commits/b4aabcc280746149c2c6f93dbd05172852dea406):
>Setting SDK_VERSION also at chatkit-core level, not ideal now as
required redundancy of the definition to what is used for publishing.
It's not a biggie though as it's only used for analytics.
Bumping up a ver no should be done as find-replace-all using the old
version. There is also a test with explicit test data for it now.

When release process is documented the info about find-replace-all as the best way to bump up the version can be added there.

Reply:
👍 Might be worth creating that release process doc now so we don't forget these details over time.

Reply:
b03bd8b80d64992df6a440a1ca4432045badd767 the existing long-standing bug is fixed and there's no need for specifying version in more than one place. I added a point to the discovery doc about documenting the release process when we'll be releasing the 1st alpha.